pkg/runtime/field_export_reconciler.go

```diff
@@ -0,0 +1,785 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package runtime
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/go-logr/logr"
+	jq "github.com/itchyny/gojq"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrlrt "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	k8sctrlutil "sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	ackv1alpha1 "github.com/aws-controllers-k8s/runtime/apis/core/v1alpha1"
+	ackcondition "github.com/aws-controllers-k8s/runtime/pkg/condition"
+	ackcfg "github.com/aws-controllers-k8s/runtime/pkg/config"
+	ackerr "github.com/aws-controllers-k8s/runtime/pkg/errors"
+	ackmetrics "github.com/aws-controllers-k8s/runtime/pkg/metrics"
+	"github.com/aws-controllers-k8s/runtime/pkg/requeue"
+	ackrtcache "github.com/aws-controllers-k8s/runtime/pkg/runtime/cache"
+	ackrtlog "github.com/aws-controllers-k8s/runtime/pkg/runtime/log"
+	acktypes "github.com/aws-controllers-k8s/runtime/pkg/types"
+)
+
+const (
+	fieldExportFinalizerString = "finalizers.services.k8s.aws/FieldExport"
+)
+
+var (
+	pathDoesNotExistError = errors.New("path does not exist in this object")
```

Comment:
Did you want to place this in [`pkg/errors`](https://github.com/aws-controllers-k8s/runtime/tree/main/pkg/errors) and export it?

Reply:
There are a number of error strings in this package. I see resource references have their own error file. Should I move everything into one like that?

Reply:
Up to you! Feel free to create a field_export.go file in pkg/errors or add to the errors.go file. Whatever you feel is best.